### DISCUSSION TITLE
Subword Regularization for ASR

### DISCUSSION THREAD

**Discussion by whrichd** [NVIDIA-NeMo/NeMo#5501] [Ideas]
I saw [this paper](https://ieeexplore.ieee.org/document/8683531) about subword regularization (using multiple subword segmentation hypotheses in training) in ASR training. It's a while ago, used in machine translation and supported in Sentencepiece. It does not look like NeMo supports this. Is there a specific reason? Does it not work well? Curious what the current sentiment is towards this.

**Comment by Rumeysakeskin** [comment]
@whrichd, 
Nemo has subword encoding-based CTC models like Citrinet. It supports both Word Piece Tokenizer (via HuggingFace) and Sentence Piece Tokenizer (via Google SentencePiece library). Actually, this scheme overcomes some issues faced with other character-based CTC models. These can improve execution speed while simultaneously reducing memory requirements. You can find more information about these from [ASR_with_Subword_Tokenization.ipynb](https://github.com/NVIDIA/NeMo/blob/main/tutorials/asr/ASR_with_Subword_Tokenization.ipynb) and [ASR_CTC_Language_Finetuning.ipynb](https://github.com/NVIDIA/NeMo/blob/main/tutorials/asr/ASR_CTC_Language_Finetuning.ipynb). Also I am [working](https://github.com/Rumeysakeskin/ASR-fine-tuning-for-low-resource-languages) on ASR for low-resource language with Citrinet. I hope this will help you.
